File: strongnum.c
#include<stdio.h>

int fact(int n)
{
    int c=1;
    for(int i=1;i<=n;i++){

        c=c*i;
    }
    // printf("%d\n",c);
    return c;
}
void main()
{

    int n=145,r=0;
    int t=n;

    while(n>0){

        int m=n%10;
        r+=fact(m);
        n/=10;
    }
    // printf("%d",r);
    if(r==t){
        printf("strong");
    }
    else{

        printf("not strong");
    }
}

File: strongnum.py
def fct(n):
    j=1
    for i in range(1,n+1):
        j*=i
    return j
def strong(n):
    d=n
    r=0
    while d>0:
        m=d%10
        r+=fct(m)
        d//=10
    print(r)
    if r==n:
        print("yess ")
    else:
        print("noo ")
def main():
    n=int(input("enter "))

    strong(n)

if __name__=="__main__":
    main()